mmv1/third_party/terraform/acctest/vcr_utils.go

```diff
@@ -436,6 +438,14 @@ func (p *frameworkTestProvider) Configure(ctx context.Context, req provider.Conf
 	}
 }
 
+// DataSources overrides the provider's DataSources function so that we can append test-specific data sources to the list of data sources on the provider.
+// This makes the data source(s) usable only in the context of acctests, and isn't available to users
+func (p *frameworkTestProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
+	ds := p.FrameworkProvider.DataSources(ctx)
+	ds = append(ds, fwprovider.NewGoogleProviderConfigPluginFrameworkDataSource) // google_provider_config_plugin_framework
```

Comment:
Pretty neat way to do this. Just to confirm, attempting to reference the datasource outside of a test will just show it as not available/expected?

Reply:
I'll manually check, but this code is only reached in the context of testing, with the entrypoint being ProtoV5ProviderFactories, which calls MuxedProviders, which calls the vcr_utils code